### PR TITLE
Add rest of ql products to level0

### PR DESCRIPTION
**Pull request by samaloney**
* Background
* Spectra
* Variance
* FlareFlag
* EnergyCalibration
* TMStatus (draft)

### CODE REVIEW

File: stixcore/products/tests/test_quicklook.py [hunk 1]
```diff
@@ -1,9 +1,43 @@
...
+
 from stixcore.data.test import test_data
-from stixcore.products.level0.quicklook import LightCurve as L0LightCurve
-from stixcore.products.level1.quicklook import LightCurve as L1LightCurve
+from stixcore.products.level0 import quicklook as qll0
+from stixcore.products.level1 import quicklook as qll1
+
+testpackets = [(test_data.tmtc.TM_21_6_30, qll0.LightCurve, 'ql-lightcurve',
+                '0659402030:00007', '0659402958:00007', 232),
+               (test_data.tmtc.TM_21_6_31, qll0.Background, 'ql-background',
```
Comment (nicHoch) [inline stixcore/products/tests/test_quicklook.py:12]: nice that the test_data.tmtc.... data is working/used

File: stixcore/products/level0/quicklook.py [hunk 2]
```diff
@@ -186,3 +189,433 @@ def __repr__(self):
...
+        if control['num_flares'].sum() > 0:
+            data['start_scet_coarse'] = packets.get_value('NIX00287')
+            data['end_scet_coarse'] = packets.get_value('NIX00287')
+
+            data['time'] = DateTime(packets.get_value('NIX00287'), packets.get_value('NIX00287'))
+            data['highest_flareflag'] = packets.get_value('NIX00289')
+            data['tm_byte_volume'] = packets.get('NIX00290')
+            data['average_z_loc'] = packets.get('NIX00291')
+            data['average_y_loc'] = packets.get('NIX00292')
+            data['processing_mask'] = packets.get('NIX00293')
```
Comment (nicHoch) [inline stixcore/products/level0/quicklook.py:613]: some space reduction due to narrow types?

**Review by nicHoch — APPROVED** [review]
looks good only some minor type suggestions and one NIX missing

File: stixcore/products/level0/quicklook.py [hunk 3]
```diff
@@ -186,3 +189,433 @@ def __repr__(self):
...
+
+        # Fixed for spectra
+        num_energies = 32
+        control['num_energies'] = num_energies
+        control['num_samples'] = packets.get_value('NIX00089')
+
+        # Due to the way packets are split up full contiguous block of detector 1-32 are not always
+        # down-linked to the ground so need to pad the array to write to table and later fits
+        total_samples = control['num_samples'].sum()
+        full, partial = divmod(total_samples, 32)
```
[the remark this answers is not in the document]
Comment (samaloney) [reply stixcore/products/level0/quicklook.py:293]: No not at all, but there's no value in the TM which specifies it (i.e energy mask) and it has alway been 32 at least so far?

File: stixcore/products/level0/quicklook.py [hunk 4]
```diff
@@ -186,3 +189,433 @@ def __repr__(self):
...
+        data = Data()
+        data['control_index'] = control_indices
+        data['time'] = time
+        data['duration'] = duration
+        data['loc_z'] = packets.get_value('NIX00283').astype(np.int16)
+        data['loc_y'] = packets.get_value('NIX00284').astype(np.int16)
+        data['thermal_index'] = packets.get_value('NIXD0061', attr='value').astype(np.uint16)
+        data['non_thermal_index'] = packets.get_value('NIXD0060', attr='value').astype(np.uint16)
+        data['location_status'] = packets.get_value('NIXD0059', attr='value').astype(np.uint16)
+
```
[the remark this answers is not in the document]
Comment (samaloney) [reply stixcore/products/level0/quicklook.py:468]: Hum don't see that NIX in the version of the ICD I'm using?

**Pull request merged**